File: app/src/main/java/com/example/borrowme/ui/borrowing/RequestsManagementActivity.java
package com.example.borrowme.ui.borrowing;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;
import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;
import com.bumptech.glide.Glide;
import com.example.borrowme.R;
import com.example.borrowme.ui.auth.LoginActivity;
import com.example.borrowme.ui.dashboard.HomeActivity;
import com.example.borrowme.ui.items.AddItemActivity;
import com.example.borrowme.ui.items.ItemDetailActivity;
import com.example.borrowme.ui.profile.MyLendingsActivity;
import com.example.borrowme.ui.profile.ProfileActivity;
import com.google.android.material.tabs.TabLayout;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FieldValue;
import com.google.firebase.firestore.FirebaseFirestore;
import java.text.SimpleDateFormat;
import java.util.ArrayList;
import java.util.Collections;
import java.util.Date;
import java.util.HashMap;
import java.util.HashSet;
import java.util.List;
import java.util.Locale;
import java.util.Map;
import java.util.Set;

public class RequestsManagementActivity extends AppCompatActivity {

    private TabLayout tabLayout;
    private RecyclerView rvRequests;

    private final FirebaseFirestore db = FirebaseFirestore.getInstance();
    private List<DocumentSnapshot> allRequests = new ArrayList<>();
    private final List<DocumentSnapshot> filteredRequests = new ArrayList<>();
    private RequestsAdapter adapter;
    private String currentTab = "PENDING";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        try {
            setContentView(R.layout.activity_requests_management);

            initViews();
            setupListeners();
            setupRecyclerView();
            seedMockRequests();
        } catch (Exception e) {
            Log.e("RequestsActivity", "Error in onCreate", e);
        }
    }

    @Override
    protected void onStart() {
        super.onStart();
        if (FirebaseAuth.getInstance().getCurrentUser() == null) {
            startActivity(new Intent(this, LoginActivity.class));
            finish();
        } else {
            loadRequests();
        }
    }

    private void initViews() {
        tabLayout = findViewById(R.id.tabLayout);
        rvRequests = findViewById(R.id.rvRequests);
    }

    private void setupListeners() {
        View backBtn = findViewById(R.id.btnBack);
        if (backBtn != null) backBtn.setOnClickListener(v -> finish());

        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                String text = tab.getText() != null ? tab.getText().toString().toUpperCase() : "PENDING";
                currentTab = "RETURN PENDING".equals(text) ? "RETURN_PENDING" : text;
                filterRequests();
            }
            @Override
            public void onTabUnselected(TabLayout.Tab tab) {}
            @Override
            public void onTabReselected(TabLayout.Tab tab) {}
        });

        View navHome = findViewById(R.id.navHome);
        if (navHome != null) navHome.setOnClickListener(v -> {
            startActivity(new Intent(this, HomeActivity.class));
            finish();
        });

        View navLendings = findViewById(R.id.navLendings);
        if (navLendings != null) navLendings.setOnClickListener(v -> {
            startActivity(new Intent(this, MyLendingsActivity.class));
            finish();
        });

        View navRequests = findViewById(R.id.navRequests);
        if (navRequests != null) navRequests.setOnClickListener(v -> loadRequests());

        View navProfile = findViewById(R.id.navProfile);
        if (navProfile != null) navProfile.setOnClickListener(v -> {
            startActivity(new Intent(this, ProfileActivity.class));
            finish();
        });

        View fabAdd = findViewById(R.id.fabAdd);
        if (fabAdd != null) fabAdd.setOnClickListener(v -> {
            startActivity(new Intent(this, AddItemActivity.class));
        });
    }

    private void setupRecyclerView() {
        rvRequests.setLayoutManager(new LinearLayoutManager(this));
        adapter = new RequestsAdapter(filteredRequests);
        rvRequests.setAdapter(adapter);
    }

    private void loadRequests() {
        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
        if (user == null) return;
        
        db.collection("requests").whereEqualTo("borrowerId", user.getUid())
            .get()
            .addOnSuccessListener(borrowerSnap -> {
                List<DocumentSnapshot> tempRequests = new ArrayList<>(borrowerSnap.getDocuments());
                db.collection("requests").whereEqualTo("lenderId", user.getUid())
                    .get()
                    .addOnSuccessListener(lenderSnap -> {
                        tempRequests.addAll(lenderSnap.getDocuments());
                        
                        // Distinct by ID
                        Map<String, DocumentSnapshot> uniqueMap = new HashMap<>();
                        for (DocumentSnapshot doc : tempRequests) {
                            uniqueMap.put(doc.getId(), doc);
                        }
                        allRequests = new ArrayList<>(uniqueMap.values());
                        filterRequests();
                    });
            })
            .addOnFailureListener(e -> Log.e("RequestsActivity", "Load error", e));
    }

    private void filterRequests() {
        filteredRequests.clear();
        for (DocumentSnapshot doc : allRequests) {
            if (currentTab.equals(doc.getString("status"))) {
                filteredRequests.add(doc);
            }
        }
        
        Collections.sort(filteredRequests, (d1, d2) -> {
            Date date1 = d1.getDate("timestamp");
            if (date1 == null) date1 = d1.getDate("requestedAt");
            if (date1 == null) date1 = new Date(0);
            
            Date date2 = d2.getDate("timestamp");
            if (date2 == null) date2 = d2.getDate("requestedAt");
            if (date2 == null) date2 = new Date(0);
            
            return date2.compareTo(date1); // Descending
        });
        
        adapter.notifyDataSetChanged();
    }

    private void updateRequestStatus(String requestId, String newStatus, String borrowerId, String lenderId) {
        db.collection("requests").document(requestId)
            .update("status", newStatus)
            .addOnSuccessListener(aVoid -> {
                Toast.makeText(this, "Request " + newStatus, Toast.LENGTH_SHORT).show();
                if ("APPROVED".equals(newStatus)) {
                    if (borrowerId != null) db.collection("users").document(borrowerId).update("activeBorrowCount", FieldValue.increment(1));
                    if (lenderId != null) db.collection("users").document(lenderId).update("activeLendCount", FieldValue.increment(1));
                }
                loadRequests();
            });
    }

    private void seedMockRequests() {
        FirebaseUser user = FirebaseAuth.getInstance().getCurrentUser();
        if (user == null) return;
        db.collection("requests").limit(1).get().addOnSuccessListener(snapshots -> { 
            if (snapshots.isEmpty()) {
                Map<String, Object> mock1 = new HashMap<>();
                mock1.put("itemId", "showcase_0");
                mock1.put("borrowerId", "dummy_id");
                mock1.put("lenderId", user.getUid());
                mock1.put("status", "PENDING");
                mock1.put("timestamp", FieldValue.serverTimestamp());
                db.collection("requests").add(mock1);
            }
        });
    }

    class RequestsAdapter extends RecyclerView.Adapter<RequestsAdapter.ViewHolder> {
        private final List<DocumentSnapshot> items;

        RequestsAdapter(List<DocumentSnapshot> items) {
            this.items = items;
        }

        @NonNull
        @Override
        public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
            View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.item_request, parent, false);
            return new ViewHolder(view);
        }

        @Override
        public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
            DocumentSnapshot req = items.get(position);
            String itemId = req.getString("itemId");
            String borrowerId = req.getString("borrowerId");
            String lenderId = req.getString("lenderId");
            String status = req.getString("status");
            if (status == null) status = "PENDING";
            Date timestamp = req.getDate("timestamp");
            if (timestamp == null) timestamp = req.getDate("requestedAt");

            if (timestamp != null && holder.tvRequestDate != null) {
                SimpleDateFormat sdf = new SimpleDateFormat("MMM d, yyyy", Locale.getDefault());
                holder.tvRequestDate.setText("Requested on: " + sdf.format(timestamp));
            }

            if (itemId != null) {
                db.collection("items").document(itemId).get().addOnSuccessListener(doc -> { 
                    if (doc.exists()) {
                        if (holder.tvItemName != null) holder.tvItemName.setText(doc.getString("title"));
                        String url = doc.getString("imageUrl");
                        if (url != null && !url.isEmpty() && holder.ivItemImage != null) {
                            Glide.with(holder.itemView.getContext()).load(url).placeholder(R.drawable.ic_launcher_foreground).into(holder.ivItemImage);
                        }
                    }
                });
            }

            FirebaseUser currentUser = FirebaseAuth.getInstance().getCurrentUser();
            String otherId = (currentUser != null && currentUser.getUid().equals(lenderId)) ? borrowerId : lenderId;
            if (otherId != null) {
                db.collection("users").document(otherId).get().addOnSuccessListener(doc -> { 
                    if (doc.exists()) {
                        String name = doc.getString("fullName");
                        if (name == null) name = "User";
                        String hostel = doc.getString("hostel");
                        if (hostel == null) hostel = "";
                        String profileImage = doc.getString("profileImage");
                        if (holder.tvUserInfo != null) {
                            holder.tvUserInfo.setText(!hostel.isEmpty() ? name + " • " + hostel : name);
                        }
                        
                        if (profileImage != null && !profileImage.isEmpty() && holder.ivUserSmall != null) {
                            Glide.with(holder.itemView.getContext())
                                .load(profileImage)
                                .circleCrop()
                                .placeholder(R.drawable.ic_person)
                                .error(R.drawable.ic_person)
                                .into(holder.ivUserSmall);
                        }
                    }
                });
            }

            if (holder.tvStatus != null) holder.tvStatus.setText(status.replace("_", " "));
            if (holder.btnApprove != null) holder.btnApprove.setVisibility(View.GONE);
            if (holder.btnReject != null) holder.btnReject.setVisibility(View.GONE);

            switch (status) {
                case "PENDING":
                    if (holder.tvStatus != null) {
                        holder.tvStatus.setTextColor(Color.parseColor("#F59E0B"));
                        holder.tvStatus.setBackgroundResource(R.drawable.bg_badge_subtle);
                    }
                    if (holder.statusIndicator != null) holder.statusIndicator.setBackgroundColor(Color.parseColor("#F59E0B"));
                    if (currentUser != null && currentUser.getUid().equals(lenderId)) {
                        if (holder.btnApprove != null) {
                            holder.btnApprove.setVisibility(View.VISIBLE);
                            holder.btnApprove.setOnClickListener(v -> updateRequestStatus(req.getId(), "APPROVED", borrowerId, lenderId));
                        }
                        if (holder.btnReject != null) {
                            holder.btnReject.setVisibility(View.VISIBLE);
                            holder.btnReject.setOnClickListener(v -> updateRequestStatus(req.getId(), "REJECTED", null, null));
                        }
                    }
                    break;
                case "APPROVED":
                    if (holder.tvStatus != null) {
                        holder.tvStatus.setTextColor(Color.parseColor("#10B981"));
                        holder.tvStatus.setBackgroundResource(R.drawable.bg_badge_subtle);
                    }
                    if (holder.statusIndicator != null) holder.statusIndicator.setBackgroundColor(Color.parseColor("#10B981"));
                    break;
                case "REJECTED":
                    if (holder.tvStatus != null) {
                        holder.tvStatus.setTextColor(Color.parseColor("#EF4444"));
                        holder.tvStatus.setBackgroundResource(R.drawable.bg_badge_subtle);
                    }
                    if (holder.statusIndicator != null) holder.statusIndicator.setBackgroundColor(Color.parseColor("#EF4444"));
                    break;
                case "RETURN_PENDING":
                    if (holder.tvStatus != null) {
                        holder.tvStatus.setTextColor(Color.parseColor("#2563EB"));
                        holder.tvStatus.setBackgroundResource(R.drawable.bg_badge_subtle);
                    }
                    if (holder.statusIndicator != null) holder.statusIndicator.setBackgroundColor(Color.parseColor("#2563EB"));
                    break;
            }
        }

        @Override
        public int getItemCount() {
            return items.size();
        }

        class ViewHolder extends RecyclerView.ViewHolder {
            final TextView tvItemName;
            final TextView tvUserInfo;
            final ImageView ivItemImage;
            final Button btnApprove;
            final Button btnReject;
            final TextView tvStatus;
            final View statusIndicator;
            final TextView tvRequestDate;
            final ImageView ivUserSmall;

            ViewHolder(View view) {
                super(view);
                tvItemName = view.findViewById(R.id.tvItemName);
                tvUserInfo = view.findViewById(R.id.tvUserInfo);
                ivItemImage = view.findViewById(R.id.ivItemImage);
                btnApprove = view.findViewById(R.id.btnApprove);
                btnReject = view.findViewById(R.id.btnReject);
                tvStatus = view.findViewById(R.id.tvStatus);
                statusIndicator = view.findViewById(R.id.statusIndicator);
                tvRequestDate = view.findViewById(R.id.tvRequestDate);
                ivUserSmall = view.findViewById(R.id.ivUserSmall);
            }
        }
    }
}

File: app/src/main/java/com/example/borrowme/ui/borrowing/RequestsManagementActivity.kt
package com.example.borrowme.ui.borrowing

import android.content.Intent
import android.graphics.Color
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.example.borrowme.R
import com.example.borrowme.ui.auth.LoginActivity
import com.example.borrowme.ui.dashboard.FeedActivity
import com.example.borrowme.ui.dashboard.HomeActivity
import com.example.borrowme.ui.profile.MyLendingsActivity
import com.example.borrowme.ui.profile.ProfileActivity
import com.example.borrowme.ui.items.AddItemActivity
import com.google.android.material.tabs.TabLayout
import com.google.firebase.auth.FirebaseAuth
import com.google.firebase.firestore.DocumentSnapshot
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.firestore.FieldValue
import com.google.firebase.firestore.Query
import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

class RequestsManagementActivity : AppCompatActivity() {

    private var tabLayout: TabLayout? = null
    private var rvRequests: RecyclerView? = null

    private val db = FirebaseFirestore.getInstance()
    private var allRequests = mutableListOf<DocumentSnapshot>()
    private var filteredRequests = mutableListOf<DocumentSnapshot>()
    private lateinit var adapter: RequestsAdapter
    private var currentTab = "PENDING"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        try {
            setContentView(R.layout.activity_requests_management)

            initViews()
            setupListeners()
            setupRecyclerView()
            seedMockRequests()
        } catch (e: Exception) {
            Log.e("RequestsActivity", "Error in onCreate", e)
        }
    }

    override fun onStart() {
        super.onStart()
        if (FirebaseAuth.getInstance().currentUser == null) {
            startActivity(Intent(this, LoginActivity::class.java))
            finish()
        } else {
            loadRequests()
        }
    }

    private fun initViews() {
        tabLayout = findViewById(R.id.tabLayout)
        rvRequests = findViewById(R.id.rvRequests)
    }

    private fun setupListeners() {
        findViewById<View>(R.id.btnBack)?.setOnClickListener { finish() }

        tabLayout?.addOnTabSelectedListener(object : TabLayout.OnTabSelectedListener {
            override fun onTabSelected(tab: TabLayout.Tab?) {
                val text = tab?.text?.toString()?.uppercase() ?: "PENDING"
                currentTab = if (text == "RETURN PENDING") "RETURN_PENDING" else text
                filterRequests()
            }
            override fun onTabUnselected(tab: TabLayout.Tab?) {}
            override fun onTabReselected(tab: TabLayout.Tab?) {}
        })

        findViewById<View>(R.id.navHome)?.setOnClickListener {
            startActivity(Intent(this, HomeActivity::class.java))
            finish()
        }
        findViewById<View>(R.id.navLendings)?.setOnClickListener {
            startActivity(Intent(this, MyLendingsActivity::class.java))
            finish()
        }
        findViewById<View>(R.id.navRequests)?.setOnClickListener {
            loadRequests()
        }
        findViewById<View>(R.id.navProfile)?.setOnClickListener {
            startActivity(Intent(this, ProfileActivity::class.java))
            finish()
        }
        findViewById<View>(R.id.fabAdd)?.setOnClickListener {
            startActivity(Intent(this, AddItemActivity::class.java))
        }
    }

    private fun setupRecyclerView() {
        rvRequests?.layoutManager = LinearLayoutManager(this)
        adapter = RequestsAdapter(filteredRequests)
        rvRequests?.adapter = adapter
    }

    private fun loadRequests() {
        val user = FirebaseAuth.getInstance().currentUser ?: return
        
        db.collection("requests").whereEqualTo("borrowerId", user.uid)
            .get()
            .addOnSuccessListener { borrowerSnap ->
                val tempRequests = borrowerSnap.documents.toMutableList()
                db.collection("requests").whereEqualTo("lenderId", user.uid)
                    .get()
                    .addOnSuccessListener { lenderSnap ->
                        tempRequests.addAll(lenderSnap.documents)
                        allRequests = tempRequests.distinctBy { it.id }.toMutableList()
                        filterRequests()
                    }
            }
            .addOnFailureListener { Log.e("RequestsActivity", "Load error", it) }
    }

    private fun filterRequests() {
        filteredRequests.clear()
        filteredRequests.addAll(allRequests.filter { it.getString("status") == currentTab })
        filteredRequests.sortByDescending { it.getDate("timestamp") ?: it.getDate("requestedAt") ?: Date(0) }
        adapter.notifyDataSetChanged()
    }

    private fun updateRequestStatus(requestId: String, newStatus: String, borrowerId: String?, lenderId: String?) {
        db.collection("requests").document(requestId)
            .update("status", newStatus)
            .addOnSuccessListener {
                Toast.makeText(this, "Request $newStatus", Toast.LENGTH_SHORT).show()
                if (newStatus == "APPROVED") {
                    if (borrowerId != null) db.collection("users").document(borrowerId).update("activeBorrowCount", FieldValue.increment(1))
                    if (lenderId != null) db.collection("users").document(lenderId).update("activeLendCount", FieldValue.increment(1))
                }
                loadRequests()
            }
    }

    private fun seedMockRequests() {
        val user = FirebaseAuth.getInstance().currentUser ?: return
        db.collection("requests").limit(1).get().addOnSuccessListener { 
            if (it.isEmpty) {
                val mock1 = hashMapOf(
                    "itemId" to "showcase_0",
                    "borrowerId" to "dummy_id",
                    "lenderId" to user.uid,
                    "status" to "PENDING",
                    "timestamp" to FieldValue.serverTimestamp()
                )
                db.collection("requests").add(mock1)
            }
        }
    }

    inner class RequestsAdapter(private val items: List<DocumentSnapshot>) :
        RecyclerView.Adapter<RequestsAdapter.ViewHolder>() {

        inner class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
            val tvItemName: TextView? = view.findViewById(R.id.tvItemName)
            val tvUserInfo: TextView? = view.findViewById(R.id.tvUserInfo)
            val ivItemImage: ImageView? = view.findViewById(R.id.ivItemImage)
            val btnApprove: Button? = view.findViewById(R.id.btnApprove)
            val btnReject: Button? = view.findViewById(R.id.btnReject)
            val tvStatus: TextView? = view.findViewById(R.id.tvStatus)
            val statusIndicator: View? = view.findViewById(R.id.statusIndicator)
            val tvRequestDate: TextView? = view.findViewById(R.id.tvRequestDate)
            val ivUserSmall: ImageView? = view.findViewById(R.id.ivUserSmall)
        }

        override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
            val view = LayoutInflater.from(parent.context).inflate(R.layout.item_request, parent, false)
            return ViewHolder(view)
        }

        override fun onBindViewHolder(holder: ViewHolder, position: Int) {
            val req = items[position]
            val itemId = req.getString("itemId")
            val borrowerId = req.getString("borrowerId")
            val lenderId = req.getString("lenderId")
            val status = req.getString("status") ?: "PENDING"
            val timestamp = req.getDate("timestamp") ?: req.getDate("requestedAt")

            if (timestamp != null && holder.tvRequestDate != null) {
                val sdf = SimpleDateFormat("MMM d, yyyy", Locale.getDefault())
                holder.tvRequestDate.text = "Requested on: ${sdf.format(timestamp)}"
            }

            if (itemId != null) {
                db.collection("items").document(itemId).get().addOnSuccessListener { 
                    if (it.exists()) {
                        holder.tvItemName?.text = it.getString("title")
                        val url = it.getString("imageUrl")
                        if (!url.isNullOrEmpty() && holder.ivItemImage != null) {
                            Glide.with(holder.itemView.context).load(url).placeholder(R.drawable.ic_launcher_foreground).into(holder.ivItemImage)
                        }
                    }
                }
            }

            val otherId = if (FirebaseAuth.getInstance().currentUser?.uid == lenderId) borrowerId else lenderId
            if (otherId != null) {
                db.collection("users").document(otherId).get().addOnSuccessListener { 
                    if (it.exists()) {
                        val name = it.getString("fullName") ?: "User"
                        val hostel = it.getString("hostel") ?: ""
                        val profileImage = it.getString("profileImage")
                        holder.tvUserInfo?.text = if (hostel.isNotEmpty()) "$name • $hostel" else name
                        
                        if (!profileImage.isNullOrEmpty() && holder.ivUserSmall != null) {
                            Glide.with(holder.itemView.context)
                                .load(profileImage)
                                .circleCrop()
                                .placeholder(R.drawable.ic_person)
                                .error(R.drawable.ic_person)
                                .into(holder.ivUserSmall)
                        }
                    }
                }
            }

            holder.tvStatus?.text = status.replace("_", " ")
            holder.btnApprove?.visibility = View.GONE
            holder.btnReject?.visibility = View.GONE

            when (status) {
                "PENDING" -> {
                    holder.tvStatus?.setTextColor(Color.parseColor("#F59E0B"))
                    holder.tvStatus?.setBackgroundResource(R.drawable.bg_badge_subtle)
                    holder.statusIndicator?.setBackgroundColor(Color.parseColor("#F59E0B"))
                    if (FirebaseAuth.getInstance().currentUser?.uid == lenderId) {
                        holder.btnApprove?.visibility = View.VISIBLE
                        holder.btnReject?.visibility = View.VISIBLE
                        holder.btnApprove?.setOnClickListener { updateRequestStatus(req.id, "APPROVED", borrowerId, lenderId) }
                        holder.btnReject?.setOnClickListener { updateRequestStatus(req.id, "REJECTED", null, null) }
                    }
                }
                "APPROVED" -> {
                    holder.tvStatus?.setTextColor(Color.parseColor("#10B981"))
                    holder.tvStatus?.setBackgroundResource(R.drawable.bg_badge_subtle)
                    holder.statusIndicator?.setBackgroundColor(Color.parseColor("#10B981"))
                }
                "REJECTED" -> {
                    holder.tvStatus?.setTextColor(Color.parseColor("#EF4444"))
                    holder.tvStatus?.setBackgroundResource(R.drawable.bg_badge_subtle)
                    holder.statusIndicator?.setBackgroundColor(Color.parseColor("#EF4444"))
                }
                "RETURN_PENDING" -> {
                    holder.tvStatus?.setTextColor(Color.parseColor("#2563EB"))
                    holder.tvStatus?.setBackgroundResource(R.drawable.bg_badge_subtle)
                    holder.statusIndicator?.setBackgroundColor(Color.parseColor("#2563EB"))
                }
            }
        }

        override fun getItemCount() = items.size
    }
}
